docs/current_docs/manuals/developer/state-functions.mdx

```diff
@@ -0,0 +1,65 @@
+---
+slug: /manuals/developer/state-functions
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "State as Functions"
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# State as Functions
+
+Object state can be exposed as a Dagger Function, without having to create a getter function explicitly. Depending on the language used, this state is exposed using struct fields (Go), object attributes (Python) or object properties (TypeScript).
+
+<Tabs groupId="language">
+<TabItem value="Go">
+Dagger only exposes a struct's public fields; private fields will not be exposed.
+
+Here's an example where one struct field is exposed as a Dagger Function, while the other is not:
+
+```go file=./snippets/state-functions/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+The [`dagger.field`](https://dagger-io.readthedocs.io/en/latest/module.html#dagger.field) descriptor is a wrapper of
+[`dataclasses.field`](https://docs.python.org/3/library/dataclasses.html#mutable-default-values). It creates a getter function for the attribute as well so that it's accessible from the Dagger API.
+
+Here's an example where one attribute is exposed as a Dagger Function, while the other is not:
+
+```python file=./snippets/state-functions/python/main.py
+```
```

Comment:
Commenting here since I can't comment on the renamed file.

Does this look right to you @helderco?

```python
    greeting: Annotated[str, Doc("The greeting to use")] = field(default="Hello")
    name: Annotated[str, Doc("Who to greet")] = "World"
```

I was expecting:

```diff
    greeting: Annotated[str, Doc("The greeting to use")] = field(default="Hello")
-    name: Annotated[str, Doc("Who to greet")] = "World"
+    name: Annotated[str, Doc("Who to greet")] = field(default="World")
```